package.json

```diff
@@ -13,19 +13,43 @@
   },
   "scripts": {
     "bootstrap": "pnpm install && pnpm -r build",
-    "start": "pnpm demo:start",
-    "demo:start": "pnpm --filter design-system-demo --filter design-system-styles --parallel --stream start",
-    "docs:start": "pnpm --filter design-system-documentation --filter design-system-styles --filter design-system-components --filter internet-header --parallel --stream start",
+    "start": "pnpm docs:start",
+    "test": "pnpm -r test",
+    "unit": "pnpm -r unit",
+    "demo:start": "pnpm --filter design-system-demo... --parallel --stream start",
+    "docs:start": "pnpm --filter design-system-documentation... --parallel --stream start",
+    "docs:headless": "pnpm --filter design-system-documentation start:headless",
+    "docs:test": "pnpm --filter design-system-documentation test",
+    "docs:e2e": "start-server-and-test docs:headless 9300 'pnpm --filter design-system-documentation e2e'",
+    "docs:e2e:watch": "start-server-and-test docs:headless 9300 'pnpm --filter design-system-documentation e2e:watch'",
     "styles:start": "pnpm --filter design-system-styles start",
+    "styles:test": "pnpm --filter design-system-styles test",
     "components:start": "pnpm --filter design-system-components start",
+    "components:test": "pnpm --filter design-system-components test",
+    "components:unit": "pnpm --filter design-system-components unit",
+    "components:unit:watch": "pnpm --filter design-system-components unit:watch",
+    "components:e2e": "start-server-and-test docs:headless 9300 'pnpm --filter design-system-components e2e'",
+    "components:e2e:watch": "start-server-and-test docs:headless 9300 'pnpm --filter design-system-components e2e:watch'",
+    "components:snapshots": "start-server-and-test docs:headless 9300 'pnpm --filter design-system-components snapshots'",
     "intranet-header:start": "pnpm --filter design-system-components-angular start",
-    "header:start": "pnpm --filter internet-header start",
+    "header:start": "pnpm --filter internet-header dev",
+    "header:test": "pnpm --filter internet-header test",
+    "header:unit": "pnpm --filter internet-header unit",
+    "header:unit:watch": "pnpm --filter internet-header unit:watch",
+    "header:e2e": "start-server-and-test docs:headless 9300 'pnpm --filter internet-header e2e'",
+    "header:e2e:watch": "start-server-and-test docs:headless 9300 'pnpm --filter internet-header e2e:watch'",
+    "header:snapshots": "start-server-and-test docs:headless 9300 'pnpm --filter internet-header snapshots'",
+    "icons:start": "pnpm --filter design-system-icons dev",
+    "icons:test": "pnpm --filter design-system-icons test",
+    "icons:unit": "pnpm --filter design-system-icons test",
+    "icons:unit:watch": "pnpm --filter design-system-icons test:watch",
```

Comment:
Every script that does not run a package script by filtering or running it recursivly, should be moved to the corresponding package itself:
e.g. `docs:e2e`, `docs:2e2:watch`, `components:e2e`, `components:e2e:watch`, `components:snapshots`, etc.

After this we should be able to run them recursivly like you did it in the `test` or `unit` script:
`"e2e": "pnpm -r e2e"`
or
`"e2e:watch": "pnpm -r e2e:watch"`

The main problems I have with the current solution are:
- We need to define the same server ports in the main package.json as well as in the packages itself.
- It's hard to change/maintain it, when you have to update it in multiple package.json files.
- I was not able to run the script `e2e` locally in the documentation package, because the tests are not working without a server, which provides the documentation. This makes the `e2e` script in the documentation package completly useless, when used standalone.
- Running all the servers on the same port, makes it impossible to run the affected scripts in parallel.